ch00git/05Collaboration.ipynb.py

```diff
@@ -354,7 +359,7 @@
 # Now commit the merged result:
 
 # %% jupyter={"outputs_hidden": false} language="bash"
-# git commit -a --no-edit # I added a No-edit for this non-interactive session. You can edit the commit if you like.
+# git commit -a --amend --no-edit # I added a No-edit for this non-interactive session. You can edit the commit if you like.
```

Comment:
why do we need the `amend` here? Shouldn't need to, right?

Reply:
We'll either need amend or provide a commit message. I thought omitting the commit message was intentional. I usually commit my conflict resolves with "Resolve conflict" message; maybe I should do that here.

Reply:
But git provides a default git message when creating a merge commit. Has git changed the default of merges that may not be trying a merge commit but something else?

Reply:
IIRC, that is the case only when there are no conflicts. I might be wrong here, but I think the default message is provided when the merge is possible without conflicts (and there is no fast forwarding). Is this the case or was it different at the time of writing the course?

Reply:
In my new installed laptop without any configuration beyond author name it does create the message

https://asciinema.org/a/5xpZElfdDxq8rsze1rHEHSBRp

Reply:
Ah, I see. I had no idea this was possible, thanks! The command in the course is failing because it is not generating conflicts at all. Would it be possible to add ➕  me to https://github.com/UCL/github-example as well? It is a bit hard to work on the git pages locally, given that all push/pull commands fail because of access issues 😨

Reply:
You've got access now... but I don't think you need it. Without looking into it in detail, I wonder if the problem is produce because `git` now requires setting the default action when pulling.

Reply:
git can really make you scratch your head sometimes. Turns out, the github-example repository had a legacy master branch (which was set as default), so whenever the lesson cloned the repo, it automatically used the master branch. But in the lessons where it creates a git repo locally, it uses the main branch. The conflicts did not show up because the commits were on 2 different branches.

Everything looks good now! 🫠